### PR TITLE
Quick sort added

### LINKED ISSUE #283: Quick sort
Like Merge Sort, QuickSort is a Divide and Conquer algorithm. It picks an element as pivot and partitions the given array around the picked pivot

**Is your feature request related to a problem? Please describe.**
It is a sorting algorithm with time complexity O(nlogn)

**Describe the solution you'd like**
I want to sort an array

**Additional context**

**Languages**

-   [x] Python: @rosse 
-   [x] Java: @chAlekhya
-   [x] C: @neha07kumari 
-   [x] C++ : @DefCon-018

### PR DESCRIPTION
**Pull request by DefCon-018**
# Description

I have added quick sort algorithm to c++ folder and also updated README.md file

Fixes #283 

## Type of change

**Please delete options that are not relevant**

# Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented in my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (Language folder or any other readme instructed to)
-   [x] My changes generates no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by DefCon-018** [comment]
@kaustubhgupta please review this PR.

File: C++/Quick_Sort.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/*
```
Comment (kaustubhgupta) [inline C++/Quick_Sort.cpp]: Remove this doc string

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Review by YashSaxena9 — CHANGES_REQUESTED** [review]
Your logic behind code looks fine, please remove the starting docstring :)

**Comment by DefCon-018** [comment]
first docstring removed

File: C++/Quick_Sort.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+Input format :
```
Comment (kaustubhgupta) [inline C++/Quick_Sort.cpp]: The whole doc string

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by DefCon-018** [comment]
Removed whole docstring

**Review by kaustubhgupta — APPROVED** [review]

**Pull request merged**